test-data/unit/check-multiple-inheritance.test

```diff
@@ -240,3 +240,241 @@ def dec(f: Callable[..., T]) -> Callable[..., T]:
+            pass
+        @overload
+        def __new__(cls, *args, **kwargs: Dict[str, Any]) -> Mixin1.Meta:
+            pass
+class Mixin2:
+    class Meta:
+        pass
+[builtins fixtures/dict.pyi]
+[out]
+main:2: error: Definition of "Meta" in base class "Mixin1" is incompatible with definition in base class "Mixin2"
+
+[case testMultipleInheritance_NestedClassAndAttrHaveSameName]
+class Mixin1:
+    class Nested1:
+        pass
+class Mixin2:
+    Nested1: str
+class A(Mixin1, Mixin2):
+    pass
+[out]
+main:6: error: Definition of "Nested1" in base class "Mixin1" is incompatible with definition in base class "Mixin2"
+
+[case testMultipleInheritance_NestedClassAndFunctionHaveSameName]
+class Mixin1:
+    class Nested1:
+        pass
+class Mixin2:
+    def Nested1(self) -> str:
+        pass
+class A(Mixin1, Mixin2):
+    pass
+[out]
+main:7: error: Definition of "Nested1" in base class "Mixin1" is incompatible with definition in base class "Mixin2"
+
+[case testMultipleInheritance_NestedClassAndRefToOtherClass]
+class Outer:
+    pass
+class Mixin1:
+    class Nested1:
+        pass
+class Mixin2:
+    Nested1 = Outer
+class A(Mixin2, Mixin1):
+    pass
+[out]
+main:8: error: Definition of "Nested1" in base class "Mixin2" is incompatible with definition in base class "Mixin1"
+
+[case testMultipleInheritance_ReferenceToSubclassesFromSameMRO]
+class A:
+    def __init__(self, arg: str) -> None:
+        pass
+class B(A):
+    pass
+class Base1:
+    NestedVar = A
+class Base2:
+    NestedVar = B
+class Combo(Base2, Base1): ...
+[out]
+
+[case testMultipleInheritance_ReferenceToSubclassesFromSameMROCustomMetaclass]
+class Metaclass(type):
+    pass
+class A(metaclass=Metaclass):
+    pass
+class B(A):
+    pass
+class Base1:
+    NestedVar = A
+class Base2:
+    NestedVar = B
+class Combo(Base2, Base1): ...
+[out]
+
+[case testMultipleInheritance_ReferenceToSubclassesFromSameMROOverloadedNew]
+from mixins import A, B
+class Base1:
+    NestedVar = A
+class Base2:
+    NestedVar = B
+class Combo(Base2, Base1): ...
+[file mixins.py]
+class A:
+    pass
+class B(A):
+    pass
+[file mixins.pyi]
+from typing import overload, Dict, Any
+class A:
+    @overload
+    def __new__(cls, *args, **kwargs: None) -> A:
+        pass
+    @overload
+    def __new__(cls, *args, **kwargs: Dict[str, Any]) -> A:
+        pass
+class B:
+    pass
+[builtins fixtures/dict.pyi]
+[out]
+main:6: error: Definition of "NestedVar" in base class "Base2" is incompatible with definition in base class "Base1"
+
+[case testMultipleInheritance_ReferenceToGenericClasses]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+class Generic1(Generic[T]):
+    pass
+class Generic2(Generic[T]):
+    pass
+class Base1:
+    Nested = Generic1
+class Base2:
+    Nested = Generic2
+class A(Base1, Base2):
+    pass
+[out]
+main:11: error: Definition of "Nested" in base class "Base1" is incompatible with definition in base class "Base2"
```

Comment:
Please also add a test with a valid use of generics (with a generic subclass of other generic class).